### DISCUSSION TITLE
How to force vcpkg to use the 64-bit version of MSVC

### DISCUSSION THREAD

**Discussion by danielschenk** [microsoft/vcpkg#35702] [Q&A]
When building on a 64 bit host, msbuild and vcpkg default to using the 32-bit version of the MSVC toolchain, at least for `v141` (which we're using). To make it extra clear: I'm talking about **the architecture of the toolchain itself, not the target architecture**.

Due to intermittent occurrences of weird issues such as [this one](https://stackoverflow.com/questions/4256524/lnk1318-unexpected-pdb-error-ok-0), I already switched all my projects to use the 64-bit toolchain by adding this hidden setting (which is not available in the VS GUI):
```xml
<PropertyGroup>
  <PreferredToolArchitecture>x64</PreferredToolArchitecture>
</PropertyGroup>
```

I already have a triplet override with below settings.
```cmake
set(VCPKG_PLATFORM_TOOLSET "v141")
set(VCPKG_DEP_INFO_OVERRIDE_VARS "v141")
```
Is there a way to force the 64 bit toolchain as well?

**Comment by StarGate-One** [comment]
This might work, I have not tried it, but some of these are triplet file settings, they can also be set in environment variables in a batch script before I call vcpkg port installs (Your mileage may vary - ymmv).

Inside some .bat or .cmd file called before invoking vcpkg:
`set VCPKG_CRT_LINKAGE=dynamic`
`set VCPKG_DEFAULT_HOST_TRIPLET=x64-windows`
`set VCPKG_DEFAULT_TRIPLET=x64-windows`
`set VCPKG_HOST_TRIPLET=x64-windows`
`set VCPKG_LIBRARY_LINKAGE=dynamic`
`set VCPKG_PLATFORM_TOOLSET=v143` --- I would guess you could set the same but use `v141` instead?
`set VCPKG_TARGET_ARCHITECTURE=x64`
`set VCPKG_TARGET_TRIPLET=x64-windows`
`set VCPKG_VISUAL_STUDIO_PATH=C:\Program Files\Microsoft Visual Studio\2022\Community`  --- this would be your VS install?
